tools/Java_Configurator_Source/src/TSDZ2_Configurator.java

```diff
@@ -70,6 +61,175 @@ public File getFile() {
+        }
+
+        @Override
+        protected void process(java.util.List<String> chunks) {
+            for (String line : chunks) {
+                appendText(line);
+            }
+        }
+
+        @Override
+        protected void done() {
+            String genericMessage = "Check output pane for details";
+            switch (resultType) {
+                case SUCCESS:
+                    JOptionPane.showMessageDialog(null, genericMessage, "Success", JOptionPane.PLAIN_MESSAGE);
+                    break;
+                case ERROR:
+                    JOptionPane.showMessageDialog(null, message == null ? genericMessage : message, "Compiling or flashing failed", JOptionPane.ERROR_MESSAGE);
+                    break;
+                case CANCEL:
+                    appendText("\nCANCELLED");
+                    break;
+            }
+        }
+
+        private void appendText(String text) {
+            textArea.append("\n" + text);
+            textArea.setCaretPosition(textArea.getDocument().getLength());
+        }
+
+        /**
+         * Get the ProcessBuilder for the current OS.
+         * Returns null if the OS is not recognized.
+         */
+        private ProcessBuilder getProcessBuilder() {
+            String OS = System.getProperty("os.name", "generic").toLowerCase(Locale.ENGLISH);
+            boolean isMac = OS.contains("mac") || OS.contains("darwin");
+
+            if (isMac || OS.contains("nux")) {
+                String shell = "bash";
+                String preCommand = "";
+
+                if (isMac) {
+                    // Extra setup for Macs: due to the lack of a built-in package manager, when running
+                    // shell scripts relying on non-builtin commands, the PATH needs to be updated to
+                    // account for the package manager (such as homebrew) or other custom setup.
+                    // Usually this is configured in the profile or rc file for the OS-default shell.
+                    // (zsh is default on newer macOS versions.)
+                    String sourceFormat = ". ~/%s &>/dev/null; ";
+                    if (System.getenv("SHELL").endsWith("/zsh")) {
+                        shell = "zsh";
+                        preCommand = String.format(sourceFormat + sourceFormat, ".zprofile", ".zshrc");
+                    } else {
+                        preCommand = String.format(sourceFormat + sourceFormat, ".bash_profile", ".bashrc");
+                    }
+                }
+
+                // Start a shell rather than running the command directly so that the PATH is set.
+                return new ProcessBuilder(
+                        "/bin/" + shell,
+                        "-c", String.format("%s sh %s %s --gui", preCommand, COMPILE_SCRIPT_SH, fileName));
+            }
+
+            if (OS.contains("win")) {
+                return new ProcessBuilder("cmd", "/c", "start", COMPILE_SCRIPT_BAT, fileName);
```

Comment:
On Windows the system terminal shows and Output pane is not getting populated.

Reply:
I'm unfortunately not set up to test this on Windows, but as long as the terminal is showing the output and lets you respond to the prompt about whether to flash, I feel like it's not a big problem. Previously on Mac or Linux (which don't display a terminal) there was absolutely no indication of what was going on, and no way to respond to prompts, so that was the main problem I wanted to address.

Reply:
This works:
`                return new ProcessBuilder(Paths.get(rootPath, COMPILE_SCRIPT_BAT).toString(), fileName);`
![image](https://github.com/emmebrusa/TSDZ2-Smart-EBike-1/assets/841061/fe8efa62-be66-4439-b5eb-56ff297e472a)
To stop, Cancel needs to be pressed twice, but that's ok.

Reply:
>... Previously on Mac or Linux (which don't display a terminal) there was absolutely no indication of what was going on, and no way to respond to prompts, so that was the main problem I wanted to address.

With these premises I would say that the modification is necessary, if it also worked on Windows it would be perfect.
@dzid26 has already taken a first step, all that's missing is the button to continue. I await the developments.

I'm planning a configurator review, I'll add it then. 
I hope soon, I first have to complete the tests to improve the efficiency of the version for stock displays.